gcp-deployment-manager/startup.sh

```diff
@@ -20,13 +20,22 @@ PROJECT_INACTIVITY_TIMEOUT="0.5h"
 PROJECT_DANGLING_TIMEOUT="168h"
 PROJECTS_ROOT="${BUILD_USER_HOME}/builds"
 
+# run init sctipt once
+[[ -f /root/init.lock ]] && exit 0
+
 # Create build-agent user with no-password sudo access
 # Forcing the uid to avoid race conditions with GCP creating project level users at the same time.
 # (Otherwise, we may run into something like "useradd: UID 1001 is not unique")
-if [[ "$(id -u ${BUILD_USER})" != "${BUILD_USER_UID}" ]]; then
+if [[ "$(id -u ${BUILD_USER})" == "" ]]; then
 	adduser --disabled-password --gecos "" --uid ${BUILD_USER_UID} ${BUILD_USER}
 	usermod -aG sudo ${BUILD_USER}
 	echo "${BUILD_USER} ALL=(ALL) NOPASSWD:ALL" >/etc/sudoers.d/101-${BUILD_USER}
+# change user id and group id if user created by gcp
+elif [[ "$(id -u ${BUILD_USER})" != "${BUILD_USER_UID}" ]]; then
+	groupmod -g ${BUILD_USER_UID} ${BUILD_USER}
+	usermod -u ${BUILD_USER_UID}  ${BUILD_USER}
+	usermod -aG sudo ${BUILD_USER}
+	echo "${BUILD_USER} ALL=(ALL) NOPASSWD:ALL" >/etc/sudoers.d/101-${BUILD_USER}
```

Comment:
As far as I can tell GCP already sets up passwordless sudo for you OOTB. This appears to be the case when provisioning VMs and using the startup scripts from https://github.com/docksal/sandbox-server/pull/41. I would consider removing this.

Reply:
Yeah I just confirmed this. Any SSH Keys included in project Metadata will have their user and public key automatically added to every VM in the project. I have the following SSH Keys listed in my Compute metadata:

- build-agent
- docksal
- whoever

I had added those as tests. If I SSH into the VM that gets provisioned, I see the following in /etc/group:

```
...
google-sudoers:x:1001:docksal,build-agent,whoever
...
```

And in /etc/sudoers.d/ there lives a file `google_sudoers`, again, automatically added by GCP. Its contents are:
```
%google-sudoers ALL=(ALL:ALL) NOPASSWD:ALL
```

So the groupmod/usermod code is indeed redundant.